src/Stream/BufferedStream.php

```diff
@@ -0,0 +1,264 @@
+<?php
+
+namespace Http\Message\Stream;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * A buffered stream allow to buffer over an existing.
+ *
+ * You should use this decorator when you want to seek over a not seekable stream.
+ * This stream is however read-only.
```

Comment:
right, looks like that. still, it would be nice to summarize how this works and what the implications are in the class doc.